docs/GettingStartedDocs/VisualStudioLinux.md

```diff
@@ -0,0 +1,132 @@
+4. Create an enclave library project by right clicking on the solution in the Solution Explorer
+   and selecting Add -> New Project -> Open Enclave TEE Project (Linux).  (If it is not
+   immediately visible, look under Installed -> Visual C++ -> Cross Platform -> Linux.)  Give it a name,
+   LinuxEnclave for example.  This will create a sample enclave with an ecall\_DoWorkInEnclave()
+   method exposed to applications, that will simply call an ocall\_DoWorkInHost() method that
+   will be implemented in the application.   In this walkthrough, we'll leave this project
+   as is for now, but afterwards you can modify it as you like.
+5. Configure the enclave project to use your Linux build environment, as you did in step 2.
+   At this point, the enclave would build, but cannot be run as the application doesn't
+   invoke it yet.
+6. Import the enclave into your application project, by right clicking on the application
+   project in the Solution Explorer and selecting Open Enclave Configuration -> Import Enclave,
+   then navigate to and select the EDL file (_YourEnclaveProjectName_.edl) in your enclave project.
+   This step will modify your application project settings and add some additional files to it,
+   including a C file named _YourEnclaveProjectName_\_host.c.  This C file contains a
+   sample\_enclave\_call() method that will load and call
+   ecall\_DoWorkInEnclave(), and also contains a sample implementation of a ocall\_DoWorkInHost()
+   method that just prints a message when called.  Although the app could be compiled and run
+   at this point, sample\_enclave\_call() is still not called from anywhere.
+7. Open the application's main.cpp (or if you are starting from another existing application,
+   whatever file you want to invoke enclave code from), and add a call to sample\_enclave\_call().
+   For example, update the main.cpp file to look like this, where the extern C declaration is needed
+   because main.cpp is a C++ file whereas the _YourEnclaveProjectName_\_host.c file is a C file:
+```C
+#include <cstdio>
+
+extern "C" {
+    void sample_enclave_call(void);
+};
+
+int main()
+{
+    printf("hello from LinuxApp!\n");
+    sample_enclave_call();
+    return 0;
+}
+```
+8. You can now set breakpoints in Visual Studio, e.g., inside ecall\_DoWorkInEnclave() and inside
+   ocall\_DoWorkInHost() and run and debug the enclave application just like any other application.
+
+The solution will have three configurations: Debug, SGX-Simulation-Debug, and Release.
+The SGX-Simulation-Debug will work the same as Debug, except that SGX support will be emulated
+rather than using hardware support.  This allows debugging on hardware that does not support SGX.
+The Debug and Release configurations can only be run (whether natively or in a VM) successfully on
+SGX-capable hardware.
+
+For the platform, use x64, since Open Enclave currently only supports 64-bit enclaves.
```

Comment:
Consider moving this line before step 8. The default architecture is x86, so the build will fail for anyone following the guide sequentially.

Reply:
Moved here and in the Windows walkthrough.